main.rb

```diff
@@ -0,0 +1,19 @@
+def bubble_sort_by(array)
+  swap = true
+  while swap
+    i = 0
+    swap = false
+    while i < (array.length - 1)
+      if yield(array[i], array[i + 1]).positive?
+        array[i], array[i + 1] = array[i + 1], array[i]
+        swap = true
+      end
+      i += 1
+    end
+  end
+  puts array
+end
+
```

Comment:
- [ ] This method is an algorithm and you need to return the array instead of printing it. 
Also, remember that Ruby methods automatically returns the last evaluated statement therefore, there won't be any need to write an explicit `return` statement. You can take advantage of that feature.

Reply:
Removed the puts/ explicit return statement.